metrotranscom/doorway
# feat: add swiping ability for the hybrid listings page on mobile

# Pull Request Template

## Issue Overview

This PR addresses #133 


## Description

- Add a way to swipe the listings list up or down from the listings hybrid page. This uses the [react-swipeable](https://www.npmjs.com/package/react-swipeable) package
- Add a way to go back to the hybrid page from either the listings list only page, or the listings map only page
- Style the "swipe area"
- Create a new layout without the footer and use that layout for the listings page as well as the search-demo page. The footer will likely not be there for mobile and might just be at the end of the listings list for desktop, so it's easier to take it out now

## How Can This Be Tested/Reviewed?

- Bring up the public site and go to the listings page.
- On desktop, verify that the listings page looks the same as before: two columns, the left is the map and is fixed and the right is the listings list and is scrollable.
- Open Chrome Developer Tools and go to Toggle Device Toolbar. Pick a mobile device 
![image](https://github.com/metrotranscom/doorway/assets/67125998/60640341-e091-45ee-bd7d-6d8cadaa22e6)
- You should see the listings combined page as above. Verify that you can still scroll through the listings list
- Swipe up on the region above the listings list. To do this on desktop, you should click on and drag that section up.
- You should now just see the listings list page like this 
![image](https://github.com/metrotranscom/doorway/assets/67125998/144c11d2-7d1c-4af5-b21c-0bd7a992aa08). Verify that you can scroll through the list and still see the swipe region at the top.
- Scroll down on the top region to go back to the combined view
- Now scroll down from the combined view, and this should take you to just the listings map page which should look like this 
![image](https://github.com/metrotranscom/doorway/assets/67125998/3bfc79e7-33f0-49a3-8416-48463e4f5c9b)
- Verify that you can still interact with the map by clicking on a pin
- Swipe up on the bottom region to go back to the listings combined page

When testing, I am not always able to get the swiping to work. I would prefer to merge this PR and test from a mobile device using the dev page.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
